airflow/models.py

```diff
@@ -3218,11 +3218,10 @@ def clear(
             # Crafting the right filter for dag_id and task_ids combo
             conditions = []
             for dag in self.subdags + [self]:
-                if dag.task_ids:
```

Comment:
This line is removed to fix a bug where when `-t TASK_REGEX` flag is specified, if the task_regex does not match any task, then dag.task_ids = [] results in conditions = [], and consequently no filters are applied.